virtcontainers/pkg/cgroups/manager.go

```diff
@@ -331,3 +331,24 @@ func (m *Manager) RemoveDevice(device string) error {
 	m.Unlock()
 	return fmt.Errorf("device %v not found in the cgroup", device)
 }
+
+func (m *Manager) UpdateCpuSets(cpuset string) error {
+	cgroups, err := m.GetCgroups()
+	if err != nil {
+		return fmt.Errorf("could not get original cgroups: err: %v", err)
+	}
+
+	m.logger().WithField("map-length", len(m.GetPaths())).Debug("EGERNST")
+
+	for key, cgroupPath := range m.GetPaths() {
+		m.logger().WithFields(logrus.Fields{
+			"key":  key,
+			"path": cgroupPath,
+		}).Debug("EGERNST")
+	}
+	m.Lock()
+	cgroups.CpusetCpus = cpuset
+	m.Unlock()
+
+	return m.Apply()
```

Comment:
This Apply fails as the cgroup manager doesn't see a path for the cpuset subsystem. I would've thought this would be created when we did the initial "New" creation for the cgroup-mgr; so need to debug.

My next step is probably to study other usages of libcontainer/cgroups, and add more unit tests to our existing code to understand what the gaps are.

Reply:
What was your conclusion here? Curious as to why the Apply was failing?